test/Microsoft.AspNetCore.SignalR.Tests/HubConnectionHandlerTestUtils/Hubs.cs

```diff
@@ -154,6 +155,88 @@ public Task SendToCaller(string message)
         {
             return Clients.Caller.SendAsync("Send", message);
         }
+
+        public async Task<string> StreamingConcat(ChannelReader<string> source)
+        {
+            var sb = new StringBuilder();
+
+            while (await source.WaitToReadAsync())
+            {
+                while (source.TryRead(out var item))
+                {
+                    sb.Append(item);
+                }
+            }
+
+            return sb.ToString();
+        }
+
+        public async Task<int> StreamingSum(ChannelReader<int> source)
+        {
+            var total = 0;
+            while (await source.WaitToReadAsync())
+            {
+                while (source.TryRead(out var item))
+                {
+                    total += item;
+                }
+            }
+            return total;
+        }
+
+        public async Task<List<object>> UploadArray(ChannelReader<object> source)
+        {
+            var results = new List<object>();
+
+            while (await source.WaitToReadAsync())
+            {
+                while (source.TryRead(out var item))
+                {
+                    results.Add(item);
+                }
+            }
+
+            return results;
+        }
+
+        public async Task<string> TestTypeCastingErrors(ChannelReader<int> source)
+        {
+            // TODO -- Smarter Exceptions, new exception types.
```

Comment:
You should file an issue for these. `TODO`s easily get lost :). It's OK to leave the `TODO` in, but there should be an issue tracking it as well.

Reply:
This one in particular is addressed. Will go and open issues.